src/components/Api.js

```diff
@@ -0,0 +1,174 @@
+export default class Api {
+  constructor(options) {
+    //constructor
+    this.baseUrl = options.baseUrl;
+    this._headers = options.headers;
+    //this._authorization = this.headers["authorization"];
+  }
+
+  getInitialCards(id) {
+    return fetch(`${this.baseUrl}/${id}`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
```

Comment:
Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```